code/game/objects/items/devices/laserpointer.dm

```diff
@@ -0,0 +1,218 @@
+					if(1)
+						//rank 1 with no protection, rank 2 with basic protection, or rank 3 with industrial protection
+						to_chat(C, "<span class='notice'>Something bright flashes in the corner of your vision.</span>")
+					if(2)
+						//rank 1 or 2 with no protection, rank 2 or 3 with basic protection, or rank 3 with industrial protection
+						//alternatively, rank 1 with minor vulnerability (like night vision goggles)
+						flick("flash", C.flash_eyes())
+						to_chat(C, "<span class='danger'>A bright light shines across your eyes!</span>")
+					if(3)
+						//rank 1 with minor vulnerability, rank 2 or 3 with no protection, or rank 3 with basic protection
+						if(prob(3 * diode.rating))
+							C.Weaken(1)
+						flick("flash", C.flash_eyes())
+						E.damage += 1
+						to_chat(C, "<span class='danger'>A bright light briefly blinds you!</span>")
+					if(4)
+						//rank 3 with no protection, or rank 2 with minor vulnerability
+						if(prob(5 * diode.rating))
+							C.Weaken(1)
+						flick("e_flash", C.flash_eyes())
+						E.damage += 2
+						to_chat(C, "<span class='danger'>A blinding light burns your eyes!</span>")
+			else
+				outmsg = "<span class='notice'>You shine the [src] at [C], but miss their eyes.</span>"
+
+	//robots and AI
+	else if(issilicon(target))
+		var/mob/living/silicon/S = target
+		//20% chance to actually hit the sensors
+		if(prob(effectchance * diode.rating))
+			flick("flash", S.flash_eyes(affect_silicon = TRUE))
+			if (prob(3 * diode.rating))
+				S.Weaken(diode.rating)
+			to_chat(S, "<span class='warning'>Your sensors were blinded by a laser!</span>")
+			outmsg = "<span class='notice'>You blind [S] by shining [src] at their sensors.</span>"
+			add_attack_logs(user,S,"Tried disabling using [src]")
+		else
+			outmsg = "<span class='notice'>You shine the [src] at [S], but miss their sensors.</span>"
+
+	//cameras
+	else if(istype(target, /obj/machinery/camera))
+		var/obj/machinery/camera/C = target
+		if(prob(effectchance * diode.rating))
+			C.emp_act(4 - diode.rating)
+			outmsg = "<span class='notice'>You shine the [src] into the lens of [C].</span>"
+			add_attack_logs(user,C.name,"Tried disabling using [src]")
+		else
+			outmsg = "<span class='info'>You missed the lens of [C] with [src].</span>"
+
+	//cats!
+	for(var/mob/living/simple_animal/cat/C in view(1,targloc))
+		if(prob(50) && !(C.client))
+			C.visible_message("<span class='warning'>[C] pounces on the light!</span>", "<span class='warning'>You pounce on the light!</span>")
+			step_towards(C, targloc)
+			C.lay_down()
+			spawn(10)
+				C.lay_down()
+		else
+			C.set_dir(get_dir(C,targloc))
+			C.visible_message("<span class='notice'>[C] watches the light.</span>", "<span class='notice'>Your attention is drawn to the mysterious glowing dot.</span>")
+
+
+	//laser pointer image
+	icon_state = "pointer_[pointer_icon_state]"
+	var/list/showto = list()
+	for(var/mob/M in range(world.view,targloc))
```

Comment:
`range()` finds things through walls and other dense objects

Reply:
Is there a better way of doing this I could use?

Reply:
`viewers()` would be more appropriate, since only the mobs that can see `src` can see `src` pointing a laser at someone